tests/bluetooth/bsim/audio/src/tbs_test.c

```diff
@@ -69,6 +109,219 @@ static struct bt_conn_cb conn_callbacks = {
+	err = bt_tbs_remote_answer(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not accept call: %d\n", err);
+		return err;
+	}
+
+	printk("Terminating call\n");
+	err = bt_tbs_terminate(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate call: %d\n", err);
+		return err;
+	}
+
+	printk("Test answer & terminate successful\n");
+
+	return err;
+}
+
+static int test_hold_retrieve(void)
+{
+	int err;
+
+	printk("%s\n", __func__);
+	err = bt_tbs_originate(0, "tel:000000000001", &call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not originate call: %d\n", err);
+		return err;
+	}
+
+	err = bt_tbs_remote_answer(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not accept call: %d\n", err);
+		return err;
+	}
+
+	printk("Holding call\n");
+	err = bt_tbs_hold(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate call: %d\n", err);
+		return err;
+	}
+
+	printk("Retrieving call\n");
+	err = bt_tbs_retrieve(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not retrieve call: %d\n", err);
+		return err;
+	}
+
+	printk("Terminating call\n");
+	err = bt_tbs_terminate(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate call: %d\n", err);
+		return err;
+	}
+
+	printk("Hold & retrieve test sucessfull\n");
+
+	return err;
+}
+
+static int test_join(void)
+{
+	int err;
+	uint8_t call_indexes[2];
+
+	printk("%s\n", __func__);
+	printk("Placing first call\n");
+	err = bt_tbs_originate(0, "tel:000000000001", &call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not originate first call: %d\n", err);
+		return err;
+	}
+
+	printk("Answering first call\n");
+	err = bt_tbs_remote_answer(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not answer first call: %d\n", err);
+		return err;
+	}
+	printk("First call answered\n");
+
+	call_indexes[0] = (uint8_t)call_index;
+
+	printk("Placing second call\n");
+	err = bt_tbs_originate(0, "tel:000000000002", &call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not originate second call: %d\n", err);
+		return err;
+	}
+
+	printk("Answering second call\n");
+	err = bt_tbs_remote_answer(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not answer second call: %d\n", err);
+		return err;
+	}
+	printk("Second call answered\n");
+
+	call_indexes[1] = (uint8_t)call_index;
+
+	printk("Joining calls\n");
+	err = bt_tbs_join(2, call_indexes);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not join calls: %d\n", err);
+		return err;
+	}
+
+	err = bt_tbs_terminate(call_indexes[0]);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate first call: %d\n", err);
+		return err;
+	}
+
+	err = bt_tbs_terminate(call_indexes[1]);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate second call: %d\n", err);
+		return err;
+	}
```

Comment:
Maybe that's a stupid question, but if the calls are joint, isn't sufficient to terminate one of them?

Reply:
I don't think the TBS states anything regard that. 

If you are in a Teams chat with 3 other people (i.e. a joined call), should you be able to terminate (i.e. hang up) on just one of the people? Maybe, maybe not. 

However, if you are in the same situation, and one of the people leaves the joined call, only 1 of the 3 calls terminates. 

So IMO it makes sense that TBS allows terminating just one call in a joined call, since the opposite (with a person leaving) does not terminate all the calls.

Reply:
Probably you have right :)